src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -299,9 +321,40 @@ protected Object setParameterValue(X x, HttpServletResponse resp, PrintWriter ou
               .setMessage(e.toString())
               .build();
             outputException(x, null, "JSON", null, error);
+          }
+      }
+    }
 
+    return arg;
+  }
 
-          }
+  protected Object setValue(X x, HttpServletResponse resp, PrintWriter out, HttpParameters p, String typeName,
+      Object object) {
+    Object arg = null;
+
+    if ( ! SafetyUtil.isEmpty(typeName) ) {
+      switch ( typeName ) {
+        case "double":
+         arg = (Double) object;
+         break;
+        case "boolean":
+         arg = (Boolean) object;
+         break;
+        case "int":
+         arg = ( (Double) object ).intValue();
+         break;
+        case "long":
+          arg = ( (Double) object ).longValue();
+          break;
+        case "java.lang.String":
+          arg = object;
+          break;
+        //TODO add the case "String[]":
+        case "foam.core.X":
+          arg = x;
+          break;
+        default:
```

Comment:
The whole reason for doing this is so that more complex objects can be passed into the SUGAR API. Can you figure out how to parse FObjects here. An example of a call that you should be able to parse is the AgentAuthService.canActAs where the second and third parameters are User objects and not simple string/int/bools .. you will need to figure out how to parse these here as well.

Reply:
We will create object based on the method type.